cloud/openstack/clients/machineservice.go

```diff
@@ -50,6 +52,28 @@ type CloudConfig struct {
 	CAFile     string `json:"ca-file"`
 }
 
+// Converts clientconfig.Cloud to CloudConfig struct
+func ToClientConfig(cloud *clientconfig.Cloud) (*CloudConfig, error) {
+	conf := new(CloudConfig)
```

Comment:
Do we still need this struct?

Considering we're moving away from the custom config file to `clouds.yaml` it might be good to just use types provided by `gophercloud` rather than keeping this struct around. For example, `CloudConfig` doesn't define project specific options, keystone API versions, etc. This is part of what this issue is supposed to fix.

Here's an example of how this is managed in `terraform-provider-openstack`: https://github.com/flaper87/terraform-provider-openstack/blob/master/openstack/config.go#L73-L118

Reply:
As an initial fix I went for more of a band-aid solution, but yeah, if functionality is the problem it makes sense to migrate to gophercloud completely.

Reply:
Functionality and format are the issue here :D Thanks a lot